src/modules/hip/kernel/tensor_mean.hpp

```diff
@@ -0,0 +1,231 @@
+    for (int threadMax = 512; threadMax >= 1; threadMax /= 2)
+    {
+        if (hipThreadIdx_x < threadMax)
+        {
+            partialRSum_smem[hipThreadIdx_x] += partialRSum_smem[hipThreadIdx_x + threadMax];
+            partialGSum_smem[hipThreadIdx_x] += partialGSum_smem[hipThreadIdx_x + threadMax];
+            partialBSum_smem[hipThreadIdx_x] += partialBSum_smem[hipThreadIdx_x + threadMax];
+        }
+        __syncthreads();
+    }
+
+    // Final store to dst
+    if (hipThreadIdx_x == 0)
+    {
+        int totalElements = roiTensorPtrSrc[hipBlockIdx_z].xywhROI.roiHeight * roiTensorPtrSrc[hipBlockIdx_z].xywhROI.roiWidth;
+        float sum = partialRSum_smem[0] + partialGSum_smem[0] + partialBSum_smem[0];
+        int idx = hipBlockIdx_z * 4;
+        dstPtr[idx] = partialRSum_smem[0] / totalElements;
+        dstPtr[idx + 1] = partialGSum_smem[0] / totalElements;
+        dstPtr[idx + 2] = partialBSum_smem[0] / totalElements;
+        dstPtr[idx + 3] = sum  / (totalElements * 3);
+    }
+}
+
+// -------------------- Set 1 - Kernel Executors --------------------
+
+template <typename T, typename U>
+RppStatus hip_exec_tensor_mean(T *srcPtr,
+                               RpptDescPtr srcDescPtr,
+                               Rpp32f *tensorMeanArr,
+                               RpptROIPtr roiTensorPtrSrc,
+                               RpptRoiType roiType,
+                               rpp::Handle& handle)
+{
+    if (roiType == RpptRoiType::LTRB)
+        hip_exec_roi_converison_ltrb_to_xywh(roiTensorPtrSrc, handle);
+
+    int globalThreads_x = (srcDescPtr->w + 7) >> 3;
+    int globalThreads_y = srcDescPtr->h;
+    int globalThreads_z = srcDescPtr->n;
+    int gridDim_x = (int) ceil((float)globalThreads_x/LOCAL_THREADS_X);
+    int gridDim_y = (int) ceil((float)globalThreads_y/LOCAL_THREADS_Y);
+    int gridDim_z = (int) ceil((float)globalThreads_z/LOCAL_THREADS_Z);
+
+    if ((srcDescPtr->c == 1) && (srcDescPtr->layout == RpptLayout::NCHW))
+    {
+        Rpp32u tensorPartialSumArrLength = gridDim_x * gridDim_y * gridDim_z;
+        U *tensorPartialSumArr;
+        tensorPartialSumArr = reinterpret_cast<U*>(handle.GetInitHandle()->mem.mgpu.scratchBufferHip.floatmem);
+        hipMemsetAsync(tensorPartialSumArr, 0, tensorPartialSumArrLength * sizeof(U), handle.GetStream());
+        hipStreamSynchronize(handle.GetStream());
+        hipLaunchKernelGGL(tensor_sum_pln1_hip,
+                           dim3(gridDim_x, gridDim_y, gridDim_z),
+                           dim3(LOCAL_THREADS_X, LOCAL_THREADS_Y, LOCAL_THREADS_Z),
+                           0,
+                           handle.GetStream(),
+                           srcPtr,
+                           make_uint2(srcDescPtr->strides.nStride, srcDescPtr->strides.hStride),
+                           tensorPartialSumArr,
+                           roiTensorPtrSrc);
+        hipStreamSynchronize(handle.GetStream());
```

Comment:
please remove the hipStreamSyncronize here and in other instances as it is not necessary.

Reply:
Done